core/src/main/java/io/questdb/cairo/mv/DefinitionFileReader.java

```diff
@@ -0,0 +1,324 @@
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.mv;
+
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.VarcharTypeDriver;
+import io.questdb.cairo.vm.MemoryCARWImpl;
+import io.questdb.cairo.vm.MemoryCMRImpl;
+import io.questdb.cairo.vm.api.MemoryCMR;
+import io.questdb.cairo.vm.api.MemoryCR;
+import io.questdb.std.BinarySequence;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.Long256;
+import io.questdb.std.Long256Acceptor;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Misc;
+import io.questdb.std.Mutable;
+import io.questdb.std.Transient;
+import io.questdb.std.Unsafe;
+import io.questdb.std.Vect;
+import io.questdb.std.str.CharSink;
+import io.questdb.std.str.LPSZ;
+import io.questdb.std.str.Utf8Sequence;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import static io.questdb.cairo.mv.DefinitionFileUtils.*;
+
+public class DefinitionFileReader implements Closeable, Mutable {
+    private final BlocksCursor blocksCursor = new BlocksCursor();
+    private final FilesFacade ff;
+    private MemoryCMR file;
+    private MemoryCR memory;
+
+    public DefinitionFileReader(FilesFacade ff) {
+        this.ff = ff;
+    }
+
+    @Override
+    public void clear() {
+        Misc.free(memory);
+        Misc.free(file);
+    }
+
+    @Override
+    public void close() throws IOException {
+        clear();
+        memory = null;
+        file = null;
+    }
+
+    public BlocksCursor getCursor() {
+        long regionLength;
+        while (true) {
+            long currentVersion = getVersionVolatile();
+            final long regionOffset = HEADER_SIZE + file.getLong(getRegionOffsetOffset(currentVersion));
+            regionLength = file.getLong(getRegionLengthOffset(currentVersion));
+
+            final long fileBaseAddress = file.getPageAddress(0);
+            final long memoryBaseAddress = memory.resize(regionLength);
+            Vect.memcpy(memoryBaseAddress, fileBaseAddress + regionOffset, regionLength);
```

Comment:
Why do you need this memcpy? Wouldn't mmapped memory be enough considering that the file is append and grow-only?